app/uk/gov/hmrc/apinotificationpull/connectors/EnhancedApiNotificationQueueConnector.scala

```diff
@@ -56,19 +57,17 @@ class EnhancedApiNotificationQueueConnector @Inject()(config: ServicesConfig, ht
   }
 
   def getNotificationBy(notificationId: String, notificationStatus: NotificationStatus.Value)(implicit hc: HeaderCarrier): Future[Either[UpstreamErrorResponse, Notification]] = {
-
     val url = s"$serviceBaseUrl/notifications/${notificationStatus.toString}/$notificationId"
     logger.debug(s"Calling get notifications by using url: $url")
     http.GET[HttpResponse](url)
       .map {
-        case r if r.status == 404 =>throw UpstreamErrorResponse("Notifications not found", 404)
-        case r if r.status == 400 =>throw UpstreamErrorResponse("Bad Request to Notifications", 400)
+        case r if List(BAD_REQUEST, NOT_FOUND).contains(r.status) => throw UpstreamErrorResponse("Notifications not found", r.status)
```

Comment:
I think that Bad Request message has been missed here. Other than that it looks great.

Reply:
Nice spot @anna-shen-hmrc I'll fix that!